pkg/vault/README.md

```diff
@@ -0,0 +1,112 @@
+## Configure Vault Kubernetes auth method in each cluster
+
+The Mesh for Data uses [HashiCorp Vault](https://www.vaultproject.io/) to provide M4D Modules running in the remote clusters with the dataset credentials to execute their tasks.
+
+This section contains the required steps to enable the modules to authenticate to Vault in order for them to retrieve the dataset credentials.
+
+Module authentication is done by configuring Vault to use [Kubernetes auth method](https://www.vaultproject.io/docs/auth/kubernetes) in each cluster. Using this method the modules can authenticate to Vault by providing their service account token. Behind the scenes Vault authenticates the token by submitting TokenReview request to the API server of the kubernetes cluster where the module is running. 
+
+**Prerequisite for multi-cluster setup:**
+
+1. Vault should have connectivity to the remote clusters to allow it to communicate to the clusters API server.
+2. An [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resource should be created in the coordinator cluster to enable communication to Vault service from the remote clusters.
+
+
+Some of the steps described below are not specific to the Mesh for Data project but rather are Vault specific and can be found in Vault related online tutorials. The steps were tested on Kind cluster version 8.28 and OpenShift Container Platform 4.5.
+
+It is assumed that Vault is deployed in the coordinator cluster in `m4d-system` namespace prior to executing the steps.
```

Comment:
It is assumed that --> the following steps assumes that 
prior to executing the steps. --->  <delete>